src/algorithms/reference-value.ts

```diff
@@ -0,0 +1,55 @@
+import { ReferenceResult } from '../types';
+const _ = require('lodash');
+
+export function ReferenceValue(
+  x: number,
+  formulae: Array<{
+    is_reference: boolean; // true if this is a reference value
+    method: string; // the method used to calculate the reference value
+    formula: string; // the formula used to calculate the reference value
+    min: number; // the minimum value of the reference value
+    max: number; // the maximum value of the reference value
+  }>
+): ReferenceResult {
+  // find is_reference = true array element
```

Comment:
With comments, we don't need to tell what the code is doing.

A good comment gives information we can't find in code. It usually tells us why we have the following code.

In this case, you can say something like "First, try to calculate using reference formulae".